src/PositionUtilities.php

```diff
@@ -26,26 +26,16 @@ public static function getRangeFromPosition($pos, $length, $text) {
      * @return LineCharacterPosition
      */
     public static function getLineCharacterPositionFromPosition($pos, $text) : LineCharacterPosition {
-        if ($pos >= \strlen($text)) {
-            $pos = \strlen($text);
+        $length = \strlen($text);
+        if ($pos >= $length) {
+            $pos = $length;
         } elseif ($pos < 0) {
             $pos = 0;
         }
 
-        $newlinePositions = [];
-        $newlinePos = -1;
-        while ($newlinePos = \strpos($text, "\n", $newlinePos + 1)) {
-            if ($newlinePos < $pos) {
-                $newlinePositions[] = $newlinePos;
-                continue;
-            }
-            break;
-        }
-
-        $lastNewline = \count($newlinePositions) - 1;
-        $char = $pos - ($lastNewline >= 0 ? $newlinePositions[$lastNewline] + 1 : 0);
-        $line = \count($newlinePositions);
-
+        $lastNewlinePos = \strrpos($text, "\n", -($length - $pos));
+        $char = $pos - ($lastNewlinePos === false ? 0 : $lastNewlinePos + 1);
```

Comment:
I don't think this works in the following case:
```php
$text = "hello\n there";
$pos = 3;
```

`$lastNewlinePos` would be `false` because there are no newlines before position 3, so `$char` would be `0` rather than the expected value of `3`.

The tests should have caught this, so in addition to updating the logic, could you add an extra assert?